Changes to prtxn, add masked genome option and optimised clustering

codeprimate123 opened the pull request:
Threshold for prtxn is no longer user defined but calculated from the input data. PEKA can now use lowercase - masked genome. Clustering is now optimised for producing best possible clusters given a desired number of clusters, although significantly slower.

JureZmrzlikar commented:
Hi @codeprimate123 , thanks for your contribution!

I have few comments:
- The directory structure changed in the past month. Before, PEKA was located in `imaps/sandbox/kmers.py `, but now it is in `src/imaps/sandbox/kmers.py `. Please modify the first commit accordingly.
- Try to name PR after the change they introduce, e.g. "Add masked genome option to PEKA" or something similar. Current name "June deploy" is not a good choice. This PR was not included in June deploy, but even if it would be "July deploy" it says nothing about what are the actual changes. When someone will browse the PR history, he will look for content, time of PR creation is already included in PR metadata.
- Try to avoid commits of type "Make linters happy". Each commit is one separate unit and should be satisfying linters by itself.

JureZmrzlikar committed:
Update kmers script

Updates include:
- Change the way prtxn are defined
- Allow using repeat masked genome
- Improve clustering of top kmers
- Optimise top motifs clustering
- Supress sklearn convergence warnings
